### ISSUE TITLE
`blackhole` sink: allow to disable summary reporting

### ISSUE THREAD

**Issue by hhromic** [vectordotdev/vector#11909]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Sometimes, _sink-less_ transforms are required/useful, for example:
* accounting of event types without further processing them.
* use a transform for logging certain conditions centrally.

I described the use-case better in issue #10115, where it was suggested to route those _terminal_ transforms to a `blackhole` sink instead, which is what we have been doing in the meantime.

However, the `blackhole` sink will periodically report summaries like this:
```
2022-03-18T18:45:23.624263Z  INFO vector::sinks::blackhole::sink: Total events collected events=0 raw_bytes_collected=0
2022-03-18T18:45:24.625033Z  INFO vector::sinks::blackhole::sink: Total events collected events=1 raw_bytes_collected=676
2022-03-18T18:45:25.624833Z  INFO vector::sinks::blackhole::sink: Total events collected events=2 raw_bytes_collected=1311
2022-03-18T18:45:26.624330Z  INFO vector::sinks::blackhole::sink: Total events collected events=3 raw_bytes_collected=1954
```

These reports are useful for debugging, but for a production pipeline they are a bit noisy.
Therefore, it would be nice to be able to disable them, which at the moment is not possible to do _elegantly_.

### Attempted Solutions

The only solution for now is to set the [`print_interval_secs`](https://vector.dev/docs/reference/configuration/sinks/blackhole/#print_interval_secs) configuration to an arbitrary large value, i.e. 100 years, so the sink effectively never print summaries to the console.

However, this solution is not very elegant, and as a minor side-effect, it always prints the first summary:
```
2022-03-18T19:04:00.187708Z  INFO vector::sinks::blackhole::sink: Total events collected events=0 raw_bytes_collected=0
```

I tried to set `print_interval_secs` to `0` to see what happens, but the Vector worker just panics:
```
thread 'vector-worker' panicked at '`period` must be non-zero.', src/sinks/blackhole/sink.rs:58:38
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Proposal

The simplest proposal is to allow for `print_interval_secs` to be set to `0` to effectively disable the summary printing.

### Version

vector 0.20.0 (x86_64-unknown-linux-gnu 2a706a3 2022-02-11)

**Comment by jszwedko** [comment]
Good idea! I opened https://github.com/vectordotdev/vector/pull/11912 . Thanks as always for the detailed issue report.
